tests/nfs/nfs_verify_pynfs.py

```diff
@@ -0,0 +1,64 @@
+from cli.ceph.ceph import Ceph
+from cli.exceptions import OperationFailedError
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """Verify file lock operation
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    config = kw.get("config")
+    nfs_nodes = ceph_cluster.get_nodes("nfs")
+    clients = ceph_cluster.get_nodes("client")
+
+    port = config.get("port", "2049")
+    version = config.get("nfs_version", "4.0")
+
+    nfs_node = nfs_nodes[0]
+
+    # Step 1: Enable nfs
+    Ceph(clients[0]).mgr.module(action="enable", module="nfs", force=True)
+
+    # Step 2: Create an NFS cluster
+    nfs_name = "cephfs-nfs"
+    nfs_server_name = nfs_node.hostname
+    Ceph(clients[0]).nfs.cluster.create(name=nfs_name, nfs_server=nfs_server_name)
+
+    # Step 3: Perform Export on clients
+    fs_name = "cephfs"
+    nfs_name = "cephfs-nfs"
+    nfs_export = "/export"
+    nfs_mount = "/mnt/nfs"
+    fs = "cephfs"
+    for client in [clients[0], clients[1]]:
+        Ceph(client).nfs.export.create(
+            fs_name=fs_name, nfs_name=nfs_name, nfs_export=nfs_export, fs=fs
+        )
+
+    # Step 4: Perform nfs mount
+    for client in clients:
+        # Create mount dirs
+        client.exec_command(cmd=f"mkdir -p {nfs_mount}", sudo=True)
+        cmd = f"mount -t nfs -o vers={version},port={port} {nfs_server_name}:{nfs_export} {nfs_mount}"
+        out, _ = client.exec_command(cmd=cmd, sudo=True)
+        if out:
+            raise OperationFailedError(f"Failed to mount nfs on {client.hostname}")
+
+    # Step 5: Create a file on Client 1
+    cmd = (
+        f"python3 -m pip install ply;cd {nfs_mount};git clone git://linux-nfs.org/~bfields/pynfs.git;cd pynfs;-- yes | "
+        f"python setup.py build;cd nfs{version};./testserver.py {nfs_server_name}:/testvolume -v --outfile "
+        f"~/pynfs.run --maketree --showomit --rundep all > /tmp/pynfs-hotfix.log"
+    )
+    try:
+        clients[0].exec_command(cmd=cmd, sudo=True)
+    except Exception as e:
```

Comment:
Need to check the error handling.
Even when testcase failed in logs, it shows "Successfully completed pynfs tests on nfs cluster"

Reply:
Noted, check added